### PR TITLE
rgw: changes to support ragweed

### PR DESCRIPTION
**Pull request by yehudasa**
- teuthology ragweed task
 - test suite yaml config for ragweed tests (upgrade, plain)

### CODE REVIEW

**Comment by yehudasa** [comment]
@cbodley this has dnm, as it needs https://github.com/ceph/ceph/pull/13645 to be merged first. It currently points at the wip-rgw-support-ragweed branch for the base version, but that needs to be changed once the other PR gets merged. Potentially we'll want to have the other PR also backported to kraken, and use kraken as the base version.

File: qa/suites/rgw/verify/tasks/rgw_ragweed.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- ragweed:
+    client.0:
+      force-branch: ceph-master
+      rgw_server: client.0
+      stages: prepare
+- ragweed:
+    client.0:
+      force-branch: ceph-master
+      rgw_server: client.0
+      stages: check
```
Comment (cbodley) [inline qa/suites/rgw/verify/tasks/rgw_ragweed.yaml]: can this not just use a single `ragweed` task with `stages: check, prepare`?  (same with `multifs/tasks/rgw_ragweed.yaml`)

File: qa/suites/rgw/upgrade/ragweed/1-install/basever.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+meta:
+- desc: install ceph/<base> latest
+tasks:
+- install:
+    branch: wip-rgw-support-ragweed
```
Comment (cbodley) [inline qa/suites/rgw/upgrade/ragweed/1-install/basever.yaml]: TODO, as you pointed out

**Comment by liewegas** [comment]
@yehudasa ping

File: qa/suites/rgw/verify/tasks/rgw_ragweed.yaml (hunk 1 above)
Comment (yehudasa) [reply qa/suites/rgw/verify/tasks/rgw_ragweed.yaml]: @cbodley we can, but this lays out the template for multi-stage runs which are more interesting.

**Comment by yehudasa** [comment]
@cbodley can we merge this?

**Comment by cbodley** [comment]
as long as it's passing, sure 👍

**Comment by yehudasa** [comment]
@cbodley passing? who said anything about passing?

**Comment by yuriw** [comment]
[wip-yuri-testing-2018-02-14-2132](https://trello.com/c/TRo1V4u8/262-wip-yuri-testing-2018-02-14-2132)

File: qa/tasks/ragweed.py [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    """
+    Run the ragweed after everything is set up.
+
+    :param ctx: Context passed to task
+    :param config: specific configuration information
+    """
+    assert isinstance(config, dict)
+    testdir = teuthology.get_testdir(ctx)
+    attrs = ["!fails_on_rgw"]
+    if not ctx.rgw.use_fastcgi:
```
Comment (cbodley) [inline qa/tasks/ragweed.py]: > AttributeError: 'Namespace' object has no attribute 'use_fastcgi'

fcgi stuff was removed from the rgw task a while ago

**Comment by cbodley** [comment]
quite a few test failures in http://pulpito.front.sepia.ceph.com/yuriw-2018-02-15_01:25:35-rgw-wip-yuri-testing-2018-02-14-2132-distro-basic-smithi/

mainly bootstrap failures from `cd /home/ubuntu/cephtest/ragweed && ./bootstrap`

**Comment by yehudasa** [comment]
@cbodley @yuriw I pushed a new ragweed version that fixes the bootstrap issue, and updated this branch to avoid the second issue that casey was pointing at (fastcgi use)

**Comment by cbodley** [comment]
http://pulpito.ceph.com/yehudasa-2018-02-20_14:01:11-rgw-wip-rgw-ragweed-distro-basic-smithi/

**Pull request merged**